tests/server_test/mod.rs

```diff
@@ -0,0 +1,343 @@
+//! Tests that exercise the webhook behavior of the triagebot server.
+//!
+//! These tests exercise the behavior of the `triagebot` server by injecting
+//! webhook events into it, and then observing its behavior and response. They
+//! involve setting up HTTP servers, launching the `triagebot` process,
+//! injecting the webhook JSON object, and validating the result.
+//!
+//! The [`TestBuilder`] is used for configuring the test, and producing a
+//! [`ServerTestCtx`], which provides access to triagebot.
+//!
+//! See [`crate::github_client`] and [`TestBuilder`] for a discussion of how
+//! to set up API handlers.
+//!
+//! These tests require that PostgreSQL is installed and available in your
+//! PATH. The test sets up a little sandbox where a fresh PostgreSQL database
+//! is created and managed.
+//!
+//! To get the webhook JSON data to inject with
+//! [`ServerTestCtx::send_webook`], I recommend recording it by first running
+//! the triagebot server against the real github.com site with the
+//! `TRIAGEBOT_TEST_RECORD` environment variable set. See the README.md file
+//! for how to set up and run triagebot against one of your own repositories.
+//!
+//! The recording will save `.json` files in the current directory of all the
+//! events received. You can then move and rename those files into the
+//! `tests/server_test` directory. You usually should modify the JSON to
+//! rename the repository to `rust-lang/rust`.
+//!
+//! At the end of the test, you should call `ctx.events.assert_eq()` to
+//! validate that the correct HTTP actions were actually performed by
+//! triagebot. If you are uncertain about what to put in there, just start
+//! with an empty list, and the error will tell you what to add.
```

Comment:
In broad strokes this seems great -- I really like the assertions -- but I am (similarly to the github client) worried about maintaining and updating the JSON files.